app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -71,7 +71,7 @@ def parse_datasource(server, datasource, config)
           :nativeid          => datasource.id,
           :ems_ref           => datasource.path
         }
-        unless config.empty? || config['value'].empty?
+        unless config.empty? || config['value'].empty? || !config['value'].is_a?(Hash)
```

Comment:
I also think the statement is a bit hard to read at the moment. 
it starts with `unless`, and then there's also the `! ` operator...
Also, the original issue reported was regarding access to the select method of String, not Hash, so the above in that context confuses me even more (but maybe it's only me :smile: )